primitives/xcm/src/lib.rs

```diff
@@ -344,3 +353,75 @@ impl<
 		}
 	}
 }
+
+// Struct that converts a given MultiLocation into a 20 bytes account id by hashing
+// with blake2_256 and taking the first 20 bytes
+pub struct Account20Hash<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 20]> + Into<[u8; 20]> + Clone> Convert<MultiLocation, AccountId>
+	for Account20Hash<AccountId>
+{
+	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
+		let hash: [u8; 32] = ("multiloc", location.borrow())
+			.borrow()
+			.using_encoded(blake2_256);
+		let mut account_id = [0u8; 20];
+		account_id.copy_from_slice(&hash[0..20]);
+		Ok(account_id.into())
+	}
+
+	fn reverse_ref(_: impl Borrow<AccountId>) -> Result<MultiLocation, ()> {
+		Err(())
```

Comment:
What's the consequence of this? The documentation (`xcm_executor/src/traits/conversion.rs`) seems to suggest this might not be ok:

```rust
/// One of `convert`/`convert_ref` and `reverse`/`reverse_ref` MUST be implemented. If possible, implement
/// `convert_ref`, since this will never result in a clone. Use `convert` when you definitely need to consume
/// the source value.
```

Reply:
We are not doing anything different than https://github.com/paritytech/polkadot/blob/91680820118832ebddd64630ff47f8eeecc73ebb/xcm/xcm-builder/src/location_conversion.rs#L25. So I am guessing this is alright